### ISSUE TITLE
False negative ProtectedMemberInFinalClass in enums

### ISSUE THREAD

**Issue by IvanAntsiferov** [arturbosch/detekt#1489]
## Expected Behavior
ProtectedMemberInFinalClass do not reports protected funcs in enums.

## Observed Behavior
ProtectedMemberInFinalClass reports protected funcs in enums. 

## Steps to Reproduce
Some sample code to reproduce this issue:
```kotlin
enum class TextTip {
    Filter {
        override fun buildView(ctx: Context) = createTextView(
            ctx,
            text = "Press here to filter items",
            marginBottom = 0,
            marginTop = 0,
            gravity = Gravity.LEFT
        )
    },
    Sorting {
        override fun buildView(ctx: Context) = createTextView(
            ctx,
            text = "Press here to sort items",
            marginBottom = 8,
            marginTop = 8,
            gravity = Gravity.RIGHT
        )
    };

    abstract fun buildView(ctx: Context): View

    protected fun createTextView(
        ctx: Context,
        text: String,
        marginTop: Int,
        marginBottom: Int,
        gravity: Int
        // some other view params
    ) = TextView(ctx).apply {
        // some code to setup view with provided params
    }
}
```
## Context
If enum func is private, it can not be used in enum constants, so declare protected func in enum is only way to provide access to func from enum constants and restrict access from outer code.

## Your Environment
* Version of detekt used: 1.0.0-RC12
* Version of Gradle used (if applicable): 3.2.1
* Operating System and version: macOS 10.12.6
* Link to your project (if it's a public repository): private repo
